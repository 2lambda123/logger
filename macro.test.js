const { default: pluginTester } = require('babel-plugin-tester');

const baseCode = `
import { createStore, createEvent, createEffect, attach, restore } from './macro';
const fx = createEffect(() => 1);
const $data = createStore(0);
const was = createEvent();
const anotherFx = attach({
  effect: fx,
  source: $data,
  mapParams: (a) => a,
})
const $has = restore(was, "");
`;

pluginTester({
  pluginName: 'effector-logger/macro',
  plugin: require('babel-plugin-macros'),
  root: __dirname,
  filename: __filename,
  babelOptions: { filename: __filename },
  snapshot: true,
  tests: {
    'add sid': {
      code: baseCode,
    },
    'add sid and loc': {
      code: baseCode,
      pluginOptions: {
        effectorLogger: {
          addLoc: true,
          addNames: true,
        },
      },
    },
  },
});
