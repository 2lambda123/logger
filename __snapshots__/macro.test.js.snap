// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`effector-logger/macro add sid and loc: add sid and loc 1`] = `

import { createStore, createEvent, createEffect, attach, restore } from './macro';
const fx = createEffect(() => 1);
const $data = createStore(0);
const was = createEvent();
const anotherFx = attach({
  effect: fx,
  source: $data,
  mapParams: (a) => a,
})
const $has = restore(was, "");

      ↓ ↓ ↓ ↓ ↓ ↓

var _effectorFileName = '/macro.test.js';
import { restore as _restore } from 'effector-logger';
import { attach as _attach } from 'effector-logger';
import { createEffect as _createEffect } from 'effector-logger';
import { createEvent as _createEvent } from 'effector-logger';
import { createStore as _createStore } from 'effector-logger';

const fx = _createEffect(() => 1, {
  loc: {
    file: _effectorFileName,
    line: 2,
    column: 11,
  },
  name: 'fx',
  sid: '-u6p3sc',
});

const $data = _createStore(0, {
  loc: {
    file: _effectorFileName,
    line: 3,
    column: 14,
  },
  name: '$data',
  sid: '8tv02c',
});

const was = _createEvent({
  loc: {
    file: _effectorFileName,
    line: 4,
    column: 12,
  },
  name: 'was',
  sid: '-3z5gxy',
});

const anotherFx = _attach({
  ɔ: {
    effect: fx,
    source: $data,
    mapParams: (a) => a,
  },
  config: {
    loc: {
      file: _effectorFileName,
      line: 5,
      column: 18,
    },
    name: 'anotherFx',
    sid: '-4ce2k9',
  },
});

const $has = _restore(was, '', {
  loc: {
    file: _effectorFileName,
    line: 10,
    column: 13,
  },
  name: '$has',
  sid: 'css95b',
});


`;

exports[`effector-logger/macro add sid: add sid 1`] = `

import { createStore, createEvent, createEffect, attach, restore } from './macro';
const fx = createEffect(() => 1);
const $data = createStore(0);
const was = createEvent();
const anotherFx = attach({
  effect: fx,
  source: $data,
  mapParams: (a) => a,
})
const $has = restore(was, "");

      ↓ ↓ ↓ ↓ ↓ ↓

import { restore as _restore } from 'effector-logger';
import { attach as _attach } from 'effector-logger';
import { createEffect as _createEffect } from 'effector-logger';
import { createEvent as _createEvent } from 'effector-logger';
import { createStore as _createStore } from 'effector-logger';

const fx = _createEffect(() => 1, {
  name: 'fx',
  sid: '-u6p3sc',
});

const $data = _createStore(0, {
  name: '$data',
  sid: '8tv02c',
});

const was = _createEvent({
  name: 'was',
  sid: '-3z5gxy',
});

const anotherFx = _attach({
  ɔ: {
    effect: fx,
    source: $data,
    mapParams: (a) => a,
  },
  config: {
    name: 'anotherFx',
    sid: '-4ce2k9',
  },
});

const $has = _restore(was, '', {
  name: '$has',
  sid: 'css95b',
});


`;
